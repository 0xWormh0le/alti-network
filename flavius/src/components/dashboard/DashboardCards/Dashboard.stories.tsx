import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DashboardCards, { DashboardCardsProps } from './DashboardCards'

export default {
  title: 'Dashboard/DashboardCards',
  component: DashboardCards
} as Meta

const Template: Story<DashboardCardsProps> = (args) => <DashboardCards {...args} />

export const Default = Template.bind({})
Default.args = {
  cards: [
    {
      groupType: 1,
      label: 'Label',
      value: 10
    },
    {
      groupType: 2,
      label: 'Label 2',
      value: 4
    }
  ]
}

export const Loading = Template.bind({})
Loading.args = {
  cards: [
    {
      groupType: 1,
      label: 'Label',
      value: 10
    }
  ],
  loading: true
}
