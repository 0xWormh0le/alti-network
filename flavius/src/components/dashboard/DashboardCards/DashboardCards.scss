@import 'styles/_shared.scss';

$CARDS_MARGIN: 10px;
$CARD_WIDTH: 140px;

.DashboardCards {
  height: 100%;
  @include flex(column, space-between);

  // top and left margins are negative because we want to compensate for the cards' individual
  // margin that is applied below via the @include card($CARDS_MARGIN)
  margin-top: -1 * $CARDS_MARGIN;
  margin-right: $CARDS_MARGIN;
  margin-bottom: -1 * $CARDS_MARGIN;
  margin-left: -1 * $CARDS_MARGIN;

  &__card {
    @include card($CARDS_MARGIN);
    width: $CARD_WIDTH;
  }

  &__card-value,
  &__card-label {
    transition: background-color $TRANSITION_DURATION ease-in-out;
  }

  &__card-value {
    @include flex(column, center, center);
    height: 90px;
    text-align: center;
  }

  &__card-label {
    @include card-label();
  }

  &__card:hover {
    > .DashboardCards__card-value {
      background-color: rgba($AZURE, 0.05);
    }

    > .DashboardCards__card-label {
      background-color: rgba($AZURE, 0.1);
    }
  }

  &__card--active,
  &__card--active:hover {
    > .DashboardCards__card-value {
      background-color: rgba($AZURE, 0.1);
    }

    > .DashboardCards__card-label {
      color: $PALE_GREY;
      background-color: $AZURE;
    }
  }
}
