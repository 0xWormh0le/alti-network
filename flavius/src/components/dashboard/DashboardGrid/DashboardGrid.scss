@import 'styles/_shared.scss';

.DashboardGrid {
  &__main {
    @include flex(row, flex-start, stretch);
  }
}
