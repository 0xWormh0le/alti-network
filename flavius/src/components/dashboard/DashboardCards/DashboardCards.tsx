import React from 'react'
import ContentLoader from 'react-content-loader'
import cx from 'classnames'

import { largeNumberDisplay } from 'util/helpers'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import './DashboardCards.scss'

export interface DashboardCard {
  label: string
  value: number
  groupType: number
}

export interface DashboardCardsProps {
  cards: DashboardCard[]
  loading?: boolean
  selectedCardGroupType: number
  onCardClick: (key: number) => void
}

export const DashboardCards: React.FC<DashboardCardsProps> = ({
  cards,
  selectedCardGroupType,
  onCardClick,
  loading
}) => (
  <div className='DashboardCards'>
    {cards.map((card) => (
      <div
        key={card.label}
        onClick={() => onCardClick(card.groupType)}
        className={cx('DashboardCards__card', {
          'DashboardCards__card--active': card.groupType === selectedCardGroupType
        })}>
        {loading ? (
          <ContentLoader
            backgroundColor='#cce8fe'
            foregroundColor='#e5f3fe'
            height={90}
            className='DashboardCards__card-value'
            uniqueKey='DashboardCards__card-value'>
            <rect x={0} y={0} width='100%' height={90} />
          </ContentLoader>
        ) : (
          <Typography variant={TypographyVariant.H1} component='div' className='DashboardCards__card-value'>
            {largeNumberDisplay(card.value)}
          </Typography>
        )}
        <Typography variant={TypographyVariant.LABEL} className='DashboardCards__card-label'>
          {card.label}
        </Typography>
      </div>
    ))}
  </div>
)

export default DashboardCards
