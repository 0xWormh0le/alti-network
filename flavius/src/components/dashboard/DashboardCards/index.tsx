import { DashboardCard as T } from './DashboardCards'

export type DashboardCard = T
export { default } from './DashboardCards'
