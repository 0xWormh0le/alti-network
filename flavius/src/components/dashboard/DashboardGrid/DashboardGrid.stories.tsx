import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DashboardGrid, { DashboardGridProps } from './DashboardGrid'
import { BrowserRouter } from 'react-router-dom'

export default {
  title: 'Dashboard/DashboardGrid',
  component: DashboardGrid
} as Meta

const Template: Story<DashboardGridProps> = (args) => (
  <BrowserRouter>
    <DashboardGrid {...args} />
  </BrowserRouter>
)

export const Default = Template.bind({})
Default.args = {
  riskTypes: [
    {
      count: 1,
      riskTypeId: 10,
      severity: 2
    }
  ],
  cards: [
    {
      groupType: 1,
      label: 'Label',
      value: 10
    },
    {
      groupType: 2,
      label: 'Label 2',
      value: 4
    }
  ]
}

export const Loading = Template.bind({})
Loading.args = {
  loading: true,
  riskTypes: [
    {
      count: 1,
      riskTypeId: 10,
      severity: 2
    }
  ],
  cards: [
    {
      groupType: 1,
      label: 'Label',
      value: 10
    },
    {
      groupType: 2,
      label: 'Label 2',
      value: 4
    }
  ]
}
