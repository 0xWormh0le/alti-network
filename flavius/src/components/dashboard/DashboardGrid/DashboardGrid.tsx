import React from 'react'
import SectionTitle from 'components/elements/SectionTitle'
import DashboardCards, { DashboardCard } from '../DashboardCards'
import RiskTypesList from '../RiskTypesList'
import UI_STRINGS from 'util/ui-strings'
import './DashboardGrid.scss'

export interface DashboardGridProps {
  cards: DashboardCard[]
  selectedCardGroupType: number
  onCardClick: (cardKey: number) => void
  riskTypes: RiskTypeSummary[]
  loading?: boolean
}

export const DashboardGrid: React.FC<DashboardGridProps> = ({
  cards,
  loading,
  selectedCardGroupType,
  onCardClick,
  riskTypes
}) => {
  const selectedCard = cards.find((card) => card.groupType === selectedCardGroupType)
  return (
    <div className='DashboardGrid'>
      <SectionTitle titleText={UI_STRINGS.DASHBOARD.RISK_SUMMARY} />
      <div className='DashboardGrid__main'>
        <DashboardCards
          cards={cards}
          selectedCardGroupType={selectedCardGroupType}
          onCardClick={onCardClick}
          loading={loading}
        />
        <RiskTypesList
          riskTypes={riskTypes}
          categoryLabel={selectedCard ? selectedCard.label : 'Risks'}
          loading={loading}
        />
      </div>
    </div>
  )
}

export default DashboardGrid
