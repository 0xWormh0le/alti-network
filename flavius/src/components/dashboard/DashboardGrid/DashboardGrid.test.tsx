import React from 'react'
import { waitForElement } from '@testing-library/react'

import { renderWithRouter } from 'test/support/helpers'
import noop from 'lodash/noop'
import { DashboardCard } from '../DashboardCards'
import { DashboardGrid, DashboardGridProps } from './DashboardGrid'

const cards: DashboardCard[] = [
  {
    label: 'Sharing Risks',
    value: 0,
    groupType: 1 // risks with riskTypeId in the range 1XXX
  },
  {
    label: 'Relationship Risks',
    value: 0,
    groupType: 2 // risks with riskTypeId in the range 2XXX
  },
  {
    label: 'Activity Risks',
    value: 0,
    groupType: 3 // risks with riskTypeId in the range 3XXX
  },
  {
    label: 'Informational Risks',
    value: 0,
    groupType: 0 // risks with riskTypeId in the range 0XXX
  }
]

const riskTypes: RiskTypeSummary[] = [
  { count: 10, severity: 5, riskTypeId: 0 },
  { count: 8, severity: 5, riskTypeId: 10 },
  { count: 4, severity: 5, riskTypeId: 1011 },
  { count: 482, severity: 7, riskTypeId: 1020 },
  { count: 1, severity: 8, riskTypeId: 1050 },
  { count: 6, severity: 6, riskTypeId: 2000 },
  { count: 3, severity: 6, riskTypeId: 2010 },
  { count: 2, severity: 6, riskTypeId: 3100 }
]

describe('DashboardGrid', () => {
  it('renders correctly', async () => {
    const props: DashboardGridProps = {
      cards,
      riskTypes,
      selectedCardGroupType: 1,
      onCardClick: noop
    }
    const { container, getByLabelText } = renderWithRouter(<DashboardGrid {...props} />)
    const showAllRisksButton = await waitForElement(() => getByLabelText('show all risks'))
    expect(container).toMatchSnapshot()
    expect(showAllRisksButton).toHaveTextContent('Show All Sharing Risks')
  })

  it('shows button label for the selected group type correctly', async () => {
    const props: DashboardGridProps = {
      cards,
      riskTypes,
      selectedCardGroupType: 3,
      onCardClick: noop
    }
    const { getByLabelText } = renderWithRouter(<DashboardGrid {...props} />)
    const showAllRisksButton = await waitForElement(() => getByLabelText('show all risks'))
    expect(showAllRisksButton).toHaveTextContent('Show All Activity Risks')
  })

  it('shows `Show All Risks` if invalid gruopType is passed', async () => {
    const props: DashboardGridProps = {
      cards,
      riskTypes,
      selectedCardGroupType: 5,
      onCardClick: noop
    }
    const { getByLabelText } = renderWithRouter(<DashboardGrid {...props} />)
    const showAllRisksButton = await waitForElement(() => getByLabelText('show all risks'))
    expect(showAllRisksButton).toHaveTextContent('Show All Risks')
  })
})
