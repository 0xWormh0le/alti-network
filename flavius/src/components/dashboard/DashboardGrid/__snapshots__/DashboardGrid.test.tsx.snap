// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardGrid renders correctly 1`] = `
<div>
  <div
    class="DashboardGrid"
  >
    <h3
      class="Typography Typography--h3 SectionTitle"
    >
      Risk Summary
    </h3>
    <div
      class="DashboardGrid__main"
    >
      <div
        class="DashboardCards"
      >
        <div
          class="DashboardCards__card DashboardCards__card--active"
        >
          <div
            class="Typography Typography--h1 DashboardCards__card-value"
          >
            0
          </div>
          <span
            class="Typography Typography--label DashboardCards__card-label"
          >
            Sharing Risks
          </span>
        </div>
        <div
          class="DashboardCards__card"
        >
          <div
            class="Typography Typography--h1 DashboardCards__card-value"
          >
            0
          </div>
          <span
            class="Typography Typography--label DashboardCards__card-label"
          >
            Relationship Risks
          </span>
        </div>
        <div
          class="DashboardCards__card"
        >
          <div
            class="Typography Typography--h1 DashboardCards__card-value"
          >
            0
          </div>
          <span
            class="Typography Typography--label DashboardCards__card-label"
          >
            Activity Risks
          </span>
        </div>
        <div
          class="DashboardCards__card"
        >
          <div
            class="Typography Typography--h1 DashboardCards__card-value"
          >
            0
          </div>
          <span
            class="Typography Typography--label DashboardCards__card-label"
          >
            Informational Risks
          </span>
        </div>
      </div>
      <div
        class="RiskTypesList"
      >
        <div
          class="RiskTypesList__table"
        >
          <div
            class="RiskTypesList__headers"
          >
            <div
              class="Typography Typography--body-small Typography--bold Typography--uppercase RiskTypesList__heading-cell RiskTypesList--severity-column"
            >
              Risk severity
            </div>
            <div
              class="Typography Typography--body-small Typography--bold Typography--uppercase RiskTypesList__heading-cell RiskTypesList--type-column"
            >
              Risk type
            </div>
            <div
              class="Typography Typography--body-small Typography--bold Typography--uppercase RiskTypesList__heading-cell RiskTypesList--count-column"
            >
              Risk count
            </div>
            <div
              class="RiskTypesList__heading-cell RiskTypesList--actions-column"
            />
          </div>
          <div
            class="RiskTypesList__rows"
          >
            <div
              class="RiskTypeRow"
            >
              <div
                class="RiskTypeRow--severity RiskTypesList--severity-column"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--high"
                  />
                </div>
              </div>
              <div
                class="RiskTypesList--type-column"
              >
                <span
                  class="Typography Typography--body Typography--semibold"
                >
                  File Shared by Link Externally - Created Over 180 Days Ago
                </span>
              </div>
              <div
                class="RiskTypesList--count-column"
              >
                <span
                  class="Typography Typography--h2"
                >
                  1
                </span>
              </div>
              <div
                class="RiskTypesList--actions-column"
              >
                <a
                  class="NavButton"
                  href="/risks?riskTypeIds=[1050]"
                >
                  <button
                    class="Button Button--secondary"
                    type="button"
                  >
                    See all
                  </button>
                </a>
              </div>
            </div>
            <div
              class="RiskTypeRow"
            >
              <div
                class="RiskTypeRow--severity RiskTypesList--severity-column"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--medium"
                  />
                </div>
              </div>
              <div
                class="RiskTypesList--type-column"
              >
                <span
                  class="Typography Typography--body Typography--semibold"
                >
                  File Shared by Link Externally - Company Owned
                </span>
              </div>
              <div
                class="RiskTypesList--count-column"
              >
                <span
                  class="Typography Typography--h2"
                >
                  482
                </span>
              </div>
              <div
                class="RiskTypesList--actions-column"
              >
                <a
                  class="NavButton"
                  href="/risks?riskTypeIds=[1020]"
                >
                  <button
                    class="Button Button--secondary"
                    type="button"
                  >
                    See all
                  </button>
                </a>
              </div>
            </div>
            <div
              class="RiskTypeRow"
            >
              <div
                class="RiskTypeRow--severity RiskTypesList--severity-column"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--medium"
                  />
                </div>
              </div>
              <div
                class="RiskTypesList--type-column"
              >
                <span
                  class="Typography Typography--body Typography--semibold"
                >
                  File Shared to Personal Email Account - Company Owned
                </span>
              </div>
              <div
                class="RiskTypesList--count-column"
              >
                <span
                  class="Typography Typography--h2"
                >
                  6
                </span>
              </div>
              <div
                class="RiskTypesList--actions-column"
              >
                <a
                  class="NavButton"
                  href="/risks?riskTypeIds=[2000]"
                >
                  <button
                    class="Button Button--secondary"
                    type="button"
                  >
                    See all
                  </button>
                </a>
              </div>
            </div>
            <div
              class="RiskTypeRow"
            >
              <div
                class="RiskTypeRow--severity RiskTypesList--severity-column"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--medium"
                  />
                </div>
              </div>
              <div
                class="RiskTypesList--type-column"
              >
                <span
                  class="Typography Typography--body Typography--semibold"
                >
                  File Shared to Personal Email Account - Created Over 180 Days Ago
                </span>
              </div>
              <div
                class="RiskTypesList--count-column"
              >
                <span
                  class="Typography Typography--h2"
                >
                  3
                </span>
              </div>
              <div
                class="RiskTypesList--actions-column"
              >
                <a
                  class="NavButton"
                  href="/risks?riskTypeIds=[2010]"
                >
                  <button
                    class="Button Button--secondary"
                    type="button"
                  >
                    See all
                  </button>
                </a>
              </div>
            </div>
            <div
              class="RiskTypeRow"
            >
              <div
                class="RiskTypeRow--severity RiskTypesList--severity-column"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--medium"
                  />
                </div>
              </div>
              <div
                class="RiskTypesList--type-column"
              >
                <span
                  class="Typography Typography--body Typography--semibold"
                >
                  Many Recent Downloads - Internal Account
                </span>
              </div>
              <div
                class="RiskTypesList--count-column"
              >
                <span
                  class="Typography Typography--h2"
                >
                  2
                </span>
              </div>
              <div
                class="RiskTypesList--actions-column"
              >
                <a
                  class="NavButton"
                  href="/risks?riskTypeIds=[3100]"
                >
                  <button
                    class="Button Button--secondary"
                    type="button"
                  >
                    See all
                  </button>
                </a>
              </div>
            </div>
            <div
              class="RiskTypeRow"
            >
              <div
                class="RiskTypeRow--severity RiskTypesList--severity-column"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--low"
                  />
                </div>
              </div>
              <div
                class="RiskTypesList--type-column"
              >
                <span
                  class="Typography Typography--body Typography--semibold"
                >
                  Most Shared Files - Company Owned
                </span>
              </div>
              <div
                class="RiskTypesList--count-column"
              >
                <span
                  class="Typography Typography--h2"
                >
                  10
                </span>
              </div>
              <div
                class="RiskTypesList--actions-column"
              >
                <a
                  class="NavButton"
                  href="/risks?riskTypeIds=[0]"
                >
                  <button
                    class="Button Button--secondary"
                    type="button"
                  >
                    See all
                  </button>
                </a>
              </div>
            </div>
            <div
              class="RiskTypeRow"
            >
              <div
                class="RiskTypeRow--severity RiskTypesList--severity-column"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--low"
                  />
                </div>
              </div>
              <div
                class="RiskTypesList--type-column"
              >
                <span
                  class="Typography Typography--body Typography--semibold"
                >
                  Most Shared Files - Externally Owned
                </span>
              </div>
              <div
                class="RiskTypesList--count-column"
              >
                <span
                  class="Typography Typography--h2"
                >
                  8
                </span>
              </div>
              <div
                class="RiskTypesList--actions-column"
              >
                <a
                  class="NavButton"
                  href="/risks?riskTypeIds=[10]"
                >
                  <button
                    class="Button Button--secondary"
                    type="button"
                  >
                    See all
                  </button>
                </a>
              </div>
            </div>
            <div
              class="RiskTypeRow"
            >
              <div
                class="RiskTypeRow--severity RiskTypesList--severity-column"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--low"
                  />
                </div>
              </div>
              <div
                class="RiskTypesList--type-column"
              >
                <span
                  class="Typography Typography--body Typography--semibold"
                >
                  Potentially Sensitive File Shared by Link Internally - Company Owned
                </span>
              </div>
              <div
                class="RiskTypesList--count-column"
              >
                <span
                  class="Typography Typography--h2"
                >
                  4
                </span>
              </div>
              <div
                class="RiskTypesList--actions-column"
              >
                <a
                  class="NavButton"
                  href="/risks?riskTypeIds=[1011]"
                >
                  <button
                    class="Button Button--secondary"
                    type="button"
                  >
                    See all
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
        <a
          class="NavButton"
          href="/risks?riskTypeIds=[1050,1020,2000,2010,3100,0,10,1011]"
        >
          <button
            aria-label="show all risks"
            class="Button RiskTypesList__show-all-button Button--primary Button--size-large"
            type="button"
          >
            Show All Sharing Risks
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
